perun-core/src/test/java/cz/metacentrum/perun/core/entry/VosManagerEntryIntegrationTest.java

```diff
@@ -854,6 +861,148 @@ public void getEnrichedVoById() throws Exception {
+		Vo memberVo = perun.getVosManagerBl().createVo(sess, new Vo(-1, "Vo2", "vo2"));
+		Member memberWithExpiration = createMemberFromExtSource(vo);
+		Member memberWithoutExpiration = perun.getMembersManagerBl().createMember(sess, memberVo, perun.getUsersManagerBl().getUserByMember(sess, memberWithExpiration));
+		perun.getMembersManagerBl().validateMember(sess, memberWithoutExpiration);
+
+		String membershipExpirationAttrName = perun.getAttributesManager().NS_MEMBER_ATTR_DEF + ":membershipExpiration";
+		String expirationValue = "2020-02-02";
+
+		AttributeDefinition attrDef = perun.getAttributesManagerBl().getAttributeDefinition(sess, membershipExpirationAttrName);
+		perun.getAttributesManagerBl().setAttribute(sess,  memberWithExpiration, new Attribute(attrDef, expirationValue));
+
+		AttributeDefinition memberOrgsAttrDef = perun.getAttributesManagerBl().getAttributeDefinition(sess, A_MEMBER_DEF_MEMBER_ORGANIZATIONS);
+		perun.getAttributesManagerBl().setAttribute(sess, memberWithExpiration, new Attribute(memberOrgsAttrDef, new ArrayList<>(List.of(vo.getShortName()))));
+		perun.getAttributesManagerBl().setAttribute(sess, memberWithoutExpiration, new Attribute(memberOrgsAttrDef, new ArrayList<>(List.of(vo.getShortName()))));
+
+		assertEquals(perun.getAttributesManagerBl().getAttribute(sess, memberWithExpiration, membershipExpirationAttrName).getValue(), expirationValue);
+		assertNull(perun.getAttributesManagerBl().getAttribute(sess, memberWithoutExpiration, membershipExpirationAttrName).getValue());
+
+		vosManagerEntry.addMemberVo(sess, vo, memberVo);
+		assertNull(perun.getAttributesManagerBl().getAttribute(sess, memberWithExpiration, membershipExpirationAttrName).getValue());
+		assertNull(perun.getAttributesManagerBl().getAttribute(sess, memberWithoutExpiration, membershipExpirationAttrName).getValue());
+	}
+
+	@Test
+	public void removeMemberVoStartsParentVoLifecycle() throws Exception {
+		System.out.println(CLASS_NAME + "removeMemberVoStartsParentVoLifecycle");
+
+		Vo vo = perun.getVosManagerBl().createVo(sess, myVo);
+		Vo memberVo = perun.getVosManagerBl().createVo(sess, new Vo(-1, "Vo2", "vo2"));
+		Member member = createMemberFromExtSource(memberVo);
+		perun.getMembersManagerBl().validateMember(sess, member);
+
+		// set expiration rules for parent vo
+		Attribute membershipExpirationRulesAttribute = perun.getAttributesManagerBl().getAttribute(sess, vo, MembersManager.membershipExpirationRulesAttributeName);
+		LinkedHashMap<String, String> expirationRules = new LinkedHashMap<>();
+		expirationRules.put(AbstractMembershipExpirationRulesModule.membershipPeriodKeyName, "01.01.");
+		membershipExpirationRulesAttribute.setValue(expirationRules);
+		perun.getAttributesManagerBl().setAttribute(sess, vo, membershipExpirationRulesAttribute);
+
+		vosManagerEntry.addMemberVo(sess, vo, memberVo);
+		Member memberInParentVo = perun.getMembersManagerBl().getMemberByUserId(sess, vo, member.getUserId());
```

Comment:
Just a quick suggestion - we could check here that his expiration attribute is empty (that parentVo's lifecycle starts only after removing memberVo).

Reply:
added